cbmc_viewer/source_language.py

```diff
@@ -0,0 +1,114 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Source file language for language dependent features."""
+
+class Language:
+    """A utility class to handle language-specific functionality.
+
+    The class acts as the main API for language-specific functionality.
+    It:
+    - Provides language constants and utility functions
+    - Acts as a super-class to all specfic language classes
+    - Provides language-specific utility functions and automatically
+      redirects to the correct language utility class
+    """
+
+    ######################################
+    # Constants
+
+    C = "C"
+    RUST = "rust"
+    ALL_LANGUAGE_NAMES = [C, RUST]
+
+    ######################################
+    # Default language handlers
+
+    _default_language = None
+    
+    @classmethod
+    def get_default_language(cls):
+        return cls._default_language
+
+    @classmethod
+    def get_default_language_or_fail(cls):
+        if cls._default_language == None:
+            raise Exception("Default language is not set.")
+        
+        return cls._default_language
+    
+    @classmethod
+    def set_default_language(cls, lang):
+        assert lang in cls.ALL_LANGUAGE_NAMES, f"Invalid language: {lang}"
+        
+        cls._default_language = cls.get_language(lang)
+
+    ######################################
+    # Utility functions
+
+    @staticmethod
+    def get_language(name):
+        'Gets the language utility class from a string name.'
+        name_map = {
+            Language.C: C,
+            Language.RUST: Rust
+        }
+        return name_map[name]
```

Comment:
if the language isn't in, do we want to return `None` or throw an exception

Reply:
This was meant to act like an enum, in which case a missing look-up shouldn't happen. I'm not sure which is better in that case (`None` is more generally usable, but more prone to bugs in Python).